android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -645,6 +771,155 @@ void createTouchScalar(ImageResolution resolution, DisplayMetrics displayMetrics
         return motionEventList;
     }
 
+    public VideoStreamingParameters getLastCachedStreamingParameters() {
+        return parameters;
+    }
+
+    private List<VideoStreamingCapability> getSupportedCapabilities(VideoStreamingCapability rootCapability){
+
+        List<VideoStreamingCapability> validCapabilities = new ArrayList<>();
+        if (rootCapability == null){
+            return null;
+        }
+
+        List<VideoStreamingCapability> allCapabilities = rootCapability.getAdditionalVideoStreamingCapabilities();
+        if (allCapabilities == null){
+            return null;
+        }
+
+        if (allCapabilities != null){
+            rootCapability.setAdditionalVideoStreamingCapabilities(null);
+            allCapabilities.add(rootCapability);
+        }
+
+        for (VideoStreamingCapability capability : allCapabilities) {
+            ImageResolution imageResolution = capability.getPreferredResolution();
+            boolean matches = false;
+            switch (determineResolutionType(imageResolution)) {
+                case SQUARE:
+                    matches = inRange(capability, this.supportedLandscapeStreamingRange) || inRange(capability, this.supportedPortraitStreamingRange);
+                    break;
+                case PORTRAIT:
+                    matches = inRange(capability, this.supportedPortraitStreamingRange);
+                    break;
+                case LANDSCAPE:
+                    matches = inRange(capability, this.supportedLandscapeStreamingRange);
+                    break;
+                default:
+                    break;
+            }
+
+            if (matches) {
+                capability.setAdditionalVideoStreamingCapabilities(null);
+                if (!validCapabilities.contains(capability)) {
+                    validCapabilities.add(capability);
+                }
+            }
+        }
+
+        return validCapabilities;
+    }
+
+    private Boolean inRange(VideoStreamingCapability capability, VideoStreamingRange range){
+        if (capability == null) {
+            return false;
+        }
+        if (range == null) {
+            return false;
```

Comment:
A Null range will indicate no restrictions on the capabilities send so it should return true in this case